common/src/main/java/com/microsoft/identity/common/internal/broker/MicrosoftAuthClient.java

```diff
@@ -140,4 +165,39 @@ private String getCurrentActiveBrokerPackageName(@NonNull final Context context)
 
         return null;
     }
+
+    /**
+     * Returns the inactive installed authenticator.
+     * - If CP is the broker, this will return authenticator.
+     * - If authenticator is the broker, this will return CP.
+     *
+     * @param context
+     * @return package name of the installed inactive authenticator, if there's any.
+     */
+    private String getInactiveBrokerPackageName(@NonNull final Context context) {
+        final String methodName = ":getInactiveBrokerPackageName";
+        String currentBroker = getCurrentActiveBrokerPackageName(context);
+
+        final String inactiveBrokerPackageName;
+        if (currentBroker.equalsIgnoreCase(AuthenticationConstants.Broker.AZURE_AUTHENTICATOR_APP_PACKAGE_NAME)) {
+            inactiveBrokerPackageName = AuthenticationConstants.Broker.COMPANY_PORTAL_APP_PACKAGE_NAME;
+        } else if (currentBroker.equalsIgnoreCase(AuthenticationConstants.Broker.COMPANY_PORTAL_APP_PACKAGE_NAME)) {
+            inactiveBrokerPackageName = AuthenticationConstants.Broker.AZURE_AUTHENTICATOR_APP_PACKAGE_NAME;
+        } else {
+            return null;
+        }
+
+        // Verify the signature to make sure that we're not binding to malicious apps.
+        final BrokerValidator validator = new BrokerValidator(context);
+        try {
+            if (validator.verifySignature(inactiveBrokerPackageName)) {
+                return inactiveBrokerPackageName;
+            }
+        } catch (Exception e) {
+            // Log. It could be that inactiveBrokerPackageName is not installed on the app, or something's wrong.
+            Logger.error(TAG + methodName, "Failed to get inactive Broker package name.", e);
```

Comment:
>e [](start = 90, length = 1)

Will the exception e have sufficient information to distinguish between the 'inactive' broker package not installed on the device vs the package signature validation failure?